ResearchKit/Common/ORKAxisView.m

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright (c) 2015, Apple Inc. All rights reserved.
+ 
+ Redistribution and use in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ 
+ 1.  Redistributions of source code must retain the above copyright notice, this
+ list of conditions and the following disclaimer.
+ 
+ 2.  Redistributions in binary form must reproduce the above copyright notice, 
+ this list of conditions and the following disclaimer in the documentation and/or 
+ other materials provided with the distribution. 
+ 
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors 
+ may be used to endorse or promote products derived from this software without 
+ specific prior written permission. No license is granted to the trademarks of 
+ the copyright holders even if such marks are included in this software. 
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE 
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL 
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, 
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+
+ 
+#import "ORKAxisView.h"
+
+
+@interface ORKAxisView ()
+
+@property (nonatomic, strong) NSMutableArray *titleLabels;
+
+@property (nonatomic) ORKGraphAxisType axisType;
+
+@end
+
+
+@implementation ORKAxisView
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        [self sharedInit];
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    if (self) {
+        [self sharedInit];
+    }
+    return self;
+}
+
+- (void)sharedInit {
+    _titleLabels = [NSMutableArray new];
+    _leftOffset = 0;
+}
+
+- (void)layoutSubviews {
+    CGFloat segmentWidth = CGRectGetWidth(self.bounds) / (self.titleLabels.count - 1);
+    CGFloat labelWidth = segmentWidth;
+    
+    CGFloat labelHeight = (self.axisType == ORKGraphAxisTypeX) ? CGRectGetHeight(self.bounds) * 0.77 : 20;
+    
+    for (NSUInteger i = 0; i < self.titleLabels.count; i++) {
+        
+        CGFloat positionX = (self.axisType == ORKGraphAxisTypeX) ? (self.leftOffset + i * segmentWidth) : 0;
+        
+        if (i == 0) {
```

Comment:
This adjustment on the first label position is specific on the initial wave of RK apps and not relevant for the general case. Should be removed.